launcher/ui/attachdialog.cpp

```diff
@@ -66,56 +70,66 @@ AttachDialog::AttachDialog(QWidget *parent, Qt::WindowFlags f)
     m_proxyModel->setSourceModel(m_model);
     m_proxyModel->setDynamicSortFilter(true);
 
-    ui.view->setModel(m_proxyModel);
+    ui->view->setModel(m_proxyModel);
     // hide state
-    ui.view->hideColumn(ProcessModel::StateColumn);
-    ui.view->sortByColumn(ProcessModel::NameColumn, Qt::AscendingOrder);
-    ui.view->setSortingEnabled(true);
+    ui->view->hideColumn(ProcessModel::StateColumn);
+    ui->view->sortByColumn(ProcessModel::NameColumn, Qt::AscendingOrder);
+    ui->view->setSortingEnabled(true);
 
-    ui.view->setEditTriggers(QAbstractItemView::NoEditTriggers);
+    ui->view->setEditTriggers(QAbstractItemView::NoEditTriggers);
 
-    ui.view->setSelectionBehavior(QAbstractItemView::SelectRows);
-    ui.view->setSelectionMode(QAbstractItemView::SingleSelection);
-    connect(ui.view->selectionModel(), SIGNAL(currentRowChanged(QModelIndex,QModelIndex)),
+    ui->view->setSelectionBehavior(QAbstractItemView::SelectRows);
+    ui->view->setSelectionMode(QAbstractItemView::SingleSelection);
+    connect(ui->view->selectionModel(), SIGNAL(currentRowChanged(QModelIndex,QModelIndex)),
             this, SIGNAL(updateButtonState()));
-    connect(ui.view->selectionModel(), SIGNAL(currentChanged(QModelIndex,QModelIndex)),
+    connect(ui->view->selectionModel(), SIGNAL(currentChanged(QModelIndex,QModelIndex)),
             this, SLOT(selectABI(QModelIndex)));
 
-    connect(ui.view, SIGNAL(activated(QModelIndex)), SIGNAL(activate()));
+    connect(ui->view, SIGNAL(activated(QModelIndex)), SIGNAL(activate()));
 
-    new SearchLineController(ui.filter, m_proxyModel);
+    new SearchLineController(ui->filter, m_proxyModel);
 
-    ui.probeBox->setModel(m_abiModel);
+    ui->probeBox->setModel(m_abiModel);
 
     QSettings settings;
-    ui.accessMode->setCurrentIndex(settings.value(QStringLiteral(
+    ui->accessMode->setCurrentIndex(settings.value(QStringLiteral(
                                                       "Launcher/AttachAccessMode")).toInt());
 
     setWindowTitle(tr("GammaRay - Attach to Process"));
     setWindowIcon(QIcon(QStringLiteral(":/gammaray/GammaRay-128x128.png")));
 
-    ui.stackedWidget->setCurrentWidget(ui.loadingLabel);
+    ui->stackedWidget->setCurrentWidget(ui->loadingLabel);
     updateProcesses();
 }
 
+AttachDialog::~AttachDialog()
+{
+    delete ui;
```

Comment:
more elegant would be using std::unique_ptr, that would avoid the need for the explicit deletion